pkg/istio/onboarding/controller_test.go

```diff
@@ -328,6 +337,8 @@ func TestSyncService(t *testing.T) {
 
 			err := c.sync()
 			assert.Nil(t, err, "sync error should be nil")
+			// Add a sleep for processing controller to work on the queue
+			time.Sleep(100 * time.Millisecond)
```

Comment:
Any better way of doing this without the sleep? Maybe looping until the process queue length is empty?